src/ChipInput.js

```diff
@@ -344,8 +344,9 @@ class ChipInput extends React.Component {
       }
     } else {
       if (this.props.dataSourceConfig) {
-        const chips = this.state.chips.filter((c) => c[this.props.dataSourceConfig.value] !== chip)
-        if (chips.length !== this.state.chips.length) {
+        const chips = this.state.chips;
+        let changed = chips.splice(i,1) // remove the chip at index i
```

Comment:
Modifying the state and then immediately calling setState() or forceUpdate() is fine. [see this thread](https://groups.google.com/forum/#!topic/reactjs/RorfQ8a1KoA)  Using splice instead of creating a copy every time is more efficient both in time and memory. I've done it a couple of times before when i'm modifying large arrays to eek out more performance.

That being said -- these arrays in this use care are small. So any potential speed up will be completely negligible. So you're right we should follow the ReactJS rule of thumb. I'm just being pedantic.